applications/DEMApplication/custom_strategies/strategies/explicit_solver_strategy.cpp

```diff
@@ -913,54 +915,66 @@ namespace Kratos {
 
         KRATOS_TRY
         ClearFEMForces();
-        ConditionsArrayType& rConditions = GetFemModelPart().GetCommunicator().LocalMesh().Conditions();
-        ProcessInfo& r_process_info = GetFemModelPart().GetProcessInfo();
-        const ProcessInfo& r_const_process_info = GetFemModelPart().GetProcessInfo();
 
+        for (ModelPart::SubModelPartsContainerType::iterator sub_model_part = GetFemModelPart().SubModelPartsBegin(); sub_model_part != GetFemModelPart().SubModelPartsEnd(); ++sub_model_part) {
 
-        struct my_tls {
-            Vector rhs_cond;
-            Vector rhs_cond_elas;
-        };
+            ModelPart& submp = *sub_model_part;
 
-        //here the my_tls is constructed in place, which is the equivalent of "private" in OpenMP
-        block_for_each(rConditions, my_tls(), [&](Condition& rCondition, my_tls& rTLS){
-            Condition::GeometryType& geom = rCondition.GetGeometry();
-            rCondition.CalculateRightHandSide(rTLS.rhs_cond, r_const_process_info);
-            DEMWall* p_wall = dynamic_cast<DEMWall*> (&(rCondition));
-            p_wall->CalculateElasticForces(rTLS.rhs_cond_elas, r_process_info);
+            ElementsArrayType& pElements = submp.GetCommunicator().LocalMesh().Elements();
+            ElementsArrayType::iterator it = pElements.ptr_begin();
```

Comment:
note that this segfaults if there are no elements!
=> I would add at least a debug check

Reply:
I think this check should be in Release too. I mean, it has a very small cost, because it is not an inner loop, and it can save a lot of time. No need to re-compile in Debug, for example. Or no need for external users to send us an email with "this is not working, guys".